### PR TITLE
Get source mappings to work on coverage

### PR DESCRIPTION
**Pull request by henrikrudstrom**
Coveralls cannot show the coverage reports because lib/qt.js is not in the repo. would be nice to have.
should be possible to use the sourcemaps somehow, but cant get the package karma-sourcemap-loader to work. 
Any idea why this does not work?

### CODE REVIEW

File: karma.conf.js [hunk 1]
```diff
@@ -4,11 +4,16 @@ module.exports = function(config) {
...
+      {
+        pattern: 'lib/qt.js.map',
+        included: false
+      },
       'tests/common.js',
       'tests/failingTests.js',
       'tests/*/**/*.js',
       { pattern: 'tests/*/**/*.qml', included: false },
-      { pattern: 'tests/*/**/*.png', included: false }
+      { pattern: 'tests/*/**/*.png', included: false },
```
Comment (ChALkeR) [inline karma.conf.js]: Unrelated

**Comment by ChALkeR** [comment]
@henrikrudstrom Test coverage is calculated by [istanbul](https://www.npmjs.com/package/istanbul) by instrumenting the tested files before the actual tests are done. The test results are then stored locally and uploaded to Coveralls from Travis.

Atm, istanbul does not support reading sourcemaps when producing the coverage output. There is a tool called [lcov-sourcemap](https://www.npmjs.com/package/lcov-sourcemap) that should be able to transform the coverage output based on sourcemaps, but it looks not actively developed and not actively used. No idea if that works.

The proper solution to this would be to instrument `src/**/*.js`, not `lib/qt.js`, and build a special instrumented version for testing. Note that there are some files that are not actually proper js — for example, the encapsulate helpers.

I was planning to deal with this after we ditch the encapsulate helpers (blocked by #51), and, perhaps, after #45.

Both of those are actully blocked by the merging being completed first.

**Comment by ChALkeR** [comment]
In the meantime, you can always view your coverage report (over a single file, regrettably) in the local `./coverage` directory =)

**Comment by henrikrudstrom** [comment]
Guess we have to live with that for now.

**Comment by ChALkeR** [comment]
I suppose that the issue behind this was fixed in #231.
Closing this as fixed.

If I am wrong and there was anything else that this PR was intended to achieve, feel free to reopen. =)
